### ISSUE TITLE
Invoke-DbaQuery throws "An attempt was made to load a program with an incorrect format. (0x8007000B)" in 32 bit PowerShell 7

### ISSUE THREAD

**Issue by ashdar** [dataplat/dbatools#8638]
### Verified issue does not already exist?

I have searched and found no existing issue

### What error did you receive?

@{PSMessageDetails=; Exception=System.Exception: An attempt was made to load a program with an incorrect format. (0x8007000B)
 ---> System.Exception: Error connecting to [thor\sql2014]: An attempt was made to load a program with an incorrect format. (0x8007000B)
 ---> System.Management.Automation.MethodInvocationException: Exception calling "ExecuteWithResults" with "1" argument(s): "Failed to connect to server thor\sql2014."
 ---> Microsoft.SqlServer.Management.Common.ConnectionFailureException: Failed to connect to server thor\sql2014.
 ---> System.BadImageFormatException: An attempt was made to load a program with an incorrect format. (0x8007000B)
   at Microsoft.Data.SqlClient.SNINativeMethodWrapper.UnmanagedIsTokenRestricted(IntPtr token, Boolean& isRestricted)
   at Microsoft.Data.Win32NativeMethods.IsTokenRestrictedWrapper(IntPtr token)
   at Microsoft.Data.ProviderBase.DbConnectionPoolIdentity.GetCurrentNative()
   at Microsoft.Data.ProviderBase.DbConnectionPoolIdentity.GetCurrent()
   at Microsoft.Data.SqlClient.SqlConnectionFactory.CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, Object poolGroupProviderInfo, DbConnectionPool pool, DbConnection owningConnection, DbConnectionOptions userOptions)
   at Microsoft.Data.ProviderBase.DbConnectionFactory.CreateNonPooledConnection(DbConnection owningConnection, DbConnectionPoolGroup poolGroup, DbConnectionOptions userOptions)
   at Microsoft.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
   at Microsoft.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at Microsoft.Data.ProviderBase.DbConnectionClosed.TryOpenConnection(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at Microsoft.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource`1 retry, SqlConnectionOverrides overrides)
   at Microsoft.Data.SqlClient.SqlConnection.Open(SqlConnectionOverrides overrides)
   at Microsoft.Data.SqlClient.SqlConnection.Open()
   at Microsoft.SqlServer.Management.Common.ConnectionManager.InternalConnectImpl()
   at Microsoft.SqlServer.Management.Common.ConnectionManager.InternalConnect()
   at Microsoft.SqlServer.Management.Common.ConnectionManager.Connect()
   --- End of inner exception stack trace ---
   at Microsoft.SqlServer.Management.Common.ConnectionManager.Connect()
   at Microsoft.SqlServer.Management.Common.ConnectionManager.PoolConnect()
   at Microsoft.SqlServer.Management.Common.ServerConnection.ExecuteWithResults(String sqlCommand, Boolean retry)
   at CallSite.Target(Closure, CallSite, Object, String)
   --- End of inner exception stack trace ---
   at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
   at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   --- End of inner exception stack trace ---
   --- End of inner exception stack trace ---; TargetObject=thor\sql2014; CategoryInfo=ConnectionError: (thor\sql2014:PSObject) [Write-Error], Exception; FullyQualifiedErrorId=dbatools_Invoke-DbaQuery,Stop-Function; ErrorDetails=An attempt was made to load a program with an incorrect format. (0x8007000B); InvocationInfo=System.Management.Automation.InvocationInfo; ScriptStackTrace=at Stop-Function, C:\Program Files (x86)\PowerShell\Modules\dbatools\1.1.142\allcommands.ps1: line 149040
at Invoke-DbaQuery<Process>, C:\Program Files (x86)\PowerShell\Modules\dbatools\1.1.142\allcommands.ps1: line 79207
at <ScriptBlock>, <No file>: line 1; PipelineIterationInfo=System.Collections.ObjectModel.ReadOnlyCollection`1[System.Int32]}

### Steps to Reproduce

```powershell
# Start a new 32 bit PowerShell 7.3 console (but this happens under 32 bit 7.2 as well). I am using Windows Terminal.
# This works OK in 64 bit PowerShell 7.3 and in 32-bit and 64-bit PowerShell 5.1. It's just 32 bit PowerShell 7 that fails.
# Then:
Import-module dbatools -Minimum 1.1.142 -passthru
# That's OK. 
# Then:
Invoke-DbaQuery -Query 'select 1' -Sqlinstance  'thor\sql2014'
# That fails, with the above-mentioned error
# This happens on at least two different computers, and has been happening since at least 1.1.133.
# Just as a check of permissions and connectivity, do this: 
import-module invoke-sqlcmd2 -passthru
Invoke-sqlcmd2 -Query 'select 1' -ServerInstance 'thor\sql2014'
# That runs as expected.

```

### Please confirm that you are running the most recent version of dbatools

1.1.142

### Other details or mentions

Just to reiterate:
This test fails under 32 bit PowerShell 7.3, but I first noticed it with 32 bit 7.2 as well. 
This test works OK in 64 bit PowerShell 7.3 and in 32-bit and 64-bit PowerShell 5.1. 
"Once upon a time", this did work OK. Perhaps back around 1.1.100 or older? It did work with the last release in the 1.0.x series (1.0.176?).

As I am filling out this form, I see from the last question (below) that .net 4.6.2 is required. Just to be sure, I did download the web installer for 4.6.2 and ran it. I received this message:
".NET Framework 4.6.2 or a later update is already installed on this computer."
So, I believe that the "4.8.04084" that is installed on that computer is sufficient.

Can I be the only person using dbatools under 32 bit PowerShell 7? It seems unlikely.

Lastly: Yes, I do need 32 bit PowerShell because I have to use the 32 bit only database drivers to get to FoxPro data in order to load it into a SQL Server database. That works OK under Powershell 5.1, but 7.x is much faster than 5.1 on my systems.

### What PowerShell host was used when producing this error

PowerShell Core (pwsh.exe)

### PowerShell Host Version

Name                           Value
----                           -----
PSVersion                      7.3.0
PSEdition                      Core
GitCommitId                    7.3.0
OS                             Microsoft Windows 10.0.19044
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

### SQL Server Edition and Build number

Microsoft SQL Server 2014 (SP3-CU4-GDR) (KB4583462) - 12.0.6433.1 (X64) 
	Oct 31 2020 02:54:45 
	Copyright (c) Microsoft Corporation
	Developer Edition (64-bit) on Windows NT 6.3 <X64> (Build 9600: ) (Hypervisor)

### .NET Framework Version

PSChildName                      Version
-----------                      -------
v2.0.50727                       2.0.50727.4927
v3.0                             3.0.30729.4926
Windows Communication Foundation 3.0.4506.4926
Windows Presentation Foundation  3.0.6920.4902
v3.5                             3.5.30729.4926
Client                           4.8.04084
Full                             4.8.04084
Client                           4.0.0.0

**Comment by ashdar** [comment]
I forgot to mention that I did find issue #8306. That person 'fixed' their problem when they realized that they were only using 32 bit PowerShell accidentally. When the switched to 64-bit Powershell, they did not experience the problem so that issue was closed. I suspect, but have not proven, that they were seeing the same thing then that I am seeing now. The difference is that I am using 32 bit PowerShell on purpose and I can't just switch to 64 bit PowerShell 7.

**Comment by potatoqualitee** [comment]
looks like i can add that to dbatools 2.0 👍🏼

**Comment by potatoqualitee** [comment]
k this is fixed in 2.0 which will be released within 2 weeks

![image](https://user-images.githubusercontent.com/8278033/207076303-0cd89589-5567-4399-9fc4-79845f972d9b.png)

**Comment by ashdar** [comment]
Thank you very much.
